src/App.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component, Fragment } from 'react'
+import Nav from './Nav'
+import NewCard from './NewCard'
+import CardList from './CardList'
+import CardListEmpty from './CardListEmpty'
+
+export default class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      path: 'cards',
+      cards: []
+    }
+    this.updateCardList = this.updateCardList.bind(this)
+  }
+
+  componentDidMount() {
+    location.hash = this.state.path
+    window.addEventListener('hashchange', () => {
+      const path = location.hash.slice(1)
+      this.setState({ path })
+    })
+    window.dispatchEvent(new Event('hashchange'))
```

Comment:
This manual even dispatch won't be needed if you get your initial state from the hash in your constructor.